tests/test_utils.py

```diff
@@ -162,65 +162,85 @@ def check_if_job_finished(job: JobDataclass):
 
 def check_if_job_runner_result_correct(job: JobDataclass):
     """Iterate over every result and check if the distribution of the measurement is correct"""
-    for i in range(len(job.results)):
-        result: ResultDataclass = job.results[i]
-        check_standard_result_data(i, job, result)
-        assert result.meta_data is not None
+
+    assert len(job.results) == 2 * len(job.deployment.programs)
+
+    for result_index in range(len(job.results)):
+        program_index = result_index // 2  # every program is expected to have 2 results
+        result: ResultDataclass = job.results[result_index]
```

Comment:
```suggestion
    for result_index, result in enumerate(job.results):
        if result.result_type not in ("COUNTS", "PROBABILITIES"):
            continue
```
Use enumerate over `range(len(...))`.
Skip unknown result types for this check! (and maybe also in the assert before, e.g., by filtering the results list beforehand)